server/apps/postgres-app/src/main/resources/META-INF/persistence.xml

```diff
@@ -34,8 +34,7 @@
         <class>org.apache.james.mailrepository.jpa.model.JPAUrl</class>
         <class>org.apache.james.mailrepository.jpa.model.JPAMail</class>
         <class>org.apache.james.rrt.jpa.model.JPARecipientRewrite</class>
-        <class>org.apache.james.sieve.jpa.model.JPASieveQuota</class>
-        <class>org.apache.james.sieve.jpa.model.JPASieveScript</class>
+        <class>org.apache.james.sieve.postgres.model.JPASieveScript</class>
```

Comment:
Suspecting that might cause your issue... I see in the code pom.xml that james-server-guice-sieve-jpa module is still imported. Would not be surprised of shenanigans here. 

Try to switch this back to jpa package and test again to see maybe?